### ISSUE TITLE
make tasks not strictly require requirements

### ISSUE THREAD

**Issue by ACharbonneau** [nih-cfde/use-case-library-build#26]
so, a valid use case could be: 
- uc
- persona
- objective
- task

that would make it so we could have "for the future" use cases where the tech team hasn't figured out what requirements there are yet, and let us iteratively build use cases rather than having to wait until they're completely fleshed out to publish.

**Comment by marisalim** [comment]
I don't think this has been implemented yet - a value in the requirements yaml header is still needed to validate the code. leaving it blank
e.g., 
```
---
title: Notify team data administrator there are Review Catalogs to review
completed:
requirements:
---
```

 gave me this error from the process.py:
```
Exception: Error: library item t-0028: parameter reqs_str did not validate, value was None
```

**Comment by ACharbonneau** [comment]
K. I'll try to fix this. will make a pr to your pr

**Comment by marisalim** [comment]
Great! Just an fyi, I added req 29 to the new tasks on my pr (seemed a reasonable req) so the website builds right now.

**Comment by ACharbonneau** [comment]
k. So while you're working on this, can you make sure it doesn't overlap with uc-0004? I had half forgotten we'd added that one. The whole set https://github.com/nih-cfde/use-case-library-build/issues/39 needs to be in the library, but we had added uc-0004 a while ago, and it's an early version of parts of the set

**Comment by marisalim** [comment]
@ACharbonneau let me know if this makes sense - 

uc-0004 is about a data administrator doing the metadata submission from C2M2 ingest to the portal. uc-0009 seems to be the Data Custodian PI checking on submissions, but not actually doing the ingest.

**Comment by marisalim** [comment]
(just an fyi - until tasks do not require requirements for the code to validate, I am adding (#41 ) a placeholder task (t-9999) and requirement (r-99999). tagging these placeholders to new use cases with tasks/requirements not yet defined means the snakemake code will validate.)

**Comment by ACharbonneau** [comment]
I think this is a coding task I need help from @ctb on. We would like to make it so that tasks validate even if they have no requirements. The use case for this (ha!) is that we want to be able to add in future use cases to be completed, where we know the basics of what we want a user to do, but not necessarily what is required. This would allow me to put in things like Epic 2 immediately, without waiting for the various tech teams to come up with solutions.

As mentioned above, currently Marisa is spoofing it by having a placeholder requirement (r-99999) that we can put on tasks so they validate, which was needed to actually get epic 2 in

**Comment by ACharbonneau** [comment]
I tried updating the code with if statements to check to see if a task had requirements and then:
- validate them if it did
- pass with no entry if it didn't

but I couldn't get it to work, and then I broke my leg. 

If that is an approach that makes any sense, I can see if my local branch still exists and push it for you to look at. Otherwise it might be easier to not try to fix my garbage attempt.

**Comment by ctb** [comment]
err, I hacked in the code to do the validation, do you want me to remove the validation code altogether?

**Comment by ctb** [comment]
sorry, I read just the e-mail notification which didn't include the later comments.

so, "if a task has requirements, they need to be valid. but it's ok if a task has no requirements."
